jdbc-sink/src/main/java/org/springframework/cloud/stream/module/jdbc/JdbcSinkConfiguration.java

```diff
@@ -0,0 +1,217 @@
+import org.springframework.integration.jdbc.JdbcMessageHandler;
+import org.springframework.integration.jdbc.SqlParameterSourceFactory;
+import org.springframework.integration.json.JsonPropertyAccessor;
+import org.springframework.jdbc.core.JdbcOperations;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.SqlParameterSource;
+import org.springframework.jdbc.datasource.init.DataSourceInitializer;
+import org.springframework.jdbc.datasource.init.ResourceDatabasePopulator;
+import org.springframework.messaging.Message;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+
+/**
+ * A module that writes its incoming payload to an RDBMS using JDBC.
+ *
+ * @author Eric Bottard
+ * @author Thomas Risberg
+ */
+@EnableBinding(Sink.class)
+@EnableConfigurationProperties(JdbcSinkProperties.class)
+public class JdbcSinkConfiguration {
+
+	private static final Logger logger = LoggerFactory.getLogger(JdbcSinkConfiguration.class);
+
+	public static final Object NOT_SET = new Object();
+
+	@Autowired(required = false)
+	private SpelExpressionParser spelExpressionParser = new SpelExpressionParser();
+
+	@Autowired
+	private BeanFactory beanFactory;
+
+	protected EvaluationContext evaluationContext;
+
+	@Autowired
+	private JdbcSinkProperties properties;
+
+	@Autowired
+	private JdbcMessageHandler msgHandler;
+
+
+	@Bean
+	public JdbcMessageHandler jdbcMessageHandler(DataSource dataSource) {
+		final MultiValueMap<String, Expression> columnExpressionVariations = new LinkedMultiValueMap<>();
+		for (Map.Entry<String, String> entry : properties.getColumns().entrySet()) {
+			String value = entry.getValue();
+			columnExpressionVariations.add(entry.getKey(), spelExpressionParser.parseExpression(value));
+			if (!value.startsWith("payload")) {
+				columnExpressionVariations.add(entry.getKey(), spelExpressionParser.parseExpression("payload." + value));
+			}
+		}
+		JdbcMessageHandler jdbcMessageHandler = new JdbcMessageHandler(dataSource,
+				generateSql(properties.getTableName(),columnExpressionVariations.keySet()));
+		jdbcMessageHandler.setSqlParameterSourceFactory(
+				new SqlParameterSourceFactory() {
+					@Override
+					public SqlParameterSource createParameterSource(Object o) {
+						if (!(o instanceof Message)) {
+							throw new IllegalArgumentException("Unable to handle type " + o.getClass().getName());
+						}
+						Message<?> message = (Message<?>) o;
+						MapSqlParameterSource parameterSource = new MapSqlParameterSource();
+						for (String key: columnExpressionVariations.keySet()) {
+							List<Expression> spels = columnExpressionVariations.get(key);
+							Object value = NOT_SET;
+							EvaluationException lastException = null;
+							for (Expression spel : spels) {
+								try {
+									value = spel.getValue(evaluationContext, message);
+								}
```

Comment:
Looks like you have missed `break` here if the first expression evaluates correctly.